examples/prometheus_exporter/main.cc

```diff
@@ -0,0 +1,168 @@
+
+  // collection of records to export
+  std::vector<metric_sdk::Record> collection;
+  auto record = get_record("counter", 1, "{label-1:v1,label2:v2,}", aggregator);
+  do_example(aggregator, collection, record, exporter);
+}
+
+void gauge_example(prom_exp::PrometheusExporter &exporter)
+{
+  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
+      new metric_sdk::GaugeAggregator<int>(metric_api::InstrumentKind::Counter));
+
+  // collection of records to export
+  std::vector<metric_sdk::Record> collection;
+  auto record = get_record("gauge", 1, "{label-1:v1,label2:v2,}", aggregator);
+  do_example(aggregator, collection, record, exporter);
+}
+
+void mmsc_example(prom_exp::PrometheusExporter &exporter)
+{
+  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<double>>(
+      new metric_sdk::MinMaxSumCountAggregator<double>(metric_api::InstrumentKind::Counter));
+
+  // collection of records to export
+  std::vector<metric_sdk::Record> collection;
+  auto record = get_record("mmsc", 1, "{label-1:v1,label2:v2,}", aggregator);
+  do_example(aggregator, collection, record, exporter);
+}
+
+void histogram_example(prom_exp::PrometheusExporter &exporter)
+{
+  std::vector<double> boundaries{10, 20, 30, 40, 50};
+  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
+      new metric_sdk::HistogramAggregator<int>(metric_api::InstrumentKind::Counter, boundaries));
+
+  // collection of records to export
+  std::vector<metric_sdk::Record> collection;
+  auto record = get_record("histogram", 1, "{label-1:v1,label2:v2,}", aggregator);
+  do_example(aggregator, collection, record, exporter);
+}
+
+void exact_example(prom_exp::PrometheusExporter &exporter)
+{
+  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
+      new metric_sdk::ExactAggregator<int>(metric_api::InstrumentKind::Counter, true));
+
+  // collection of records to export
+  std::vector<metric_sdk::Record> collection;
+  auto record = get_record("exact", 1, "{label-1:v1,label2:v2,}", aggregator);
+  do_example(aggregator, collection, record, exporter);
+}
+
+void sketch_example(prom_exp::PrometheusExporter &exporter)
+{
+  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
+      new metric_sdk::SketchAggregator<int>(metric_api::InstrumentKind::Counter, true));
+
+  // collection of records to export
+  std::vector<metric_sdk::Record> collection;
+  auto record = get_record("sketch", 1, "{label-1:v1,label2:v2,}", aggregator);
+  do_example(aggregator, collection, record, exporter);
+}
+
+int main()
+{
+  std::string address = "localhost:8080";
+  prom_exp::PrometheusExporter exporter{address};
+
+  std::cout << "Prometheus Exporter example program running on " << address << "..." << std::endl;
+  std::cout << "Choose an aggregator example: {counter, gauge, mmsc, histogram, exact, sketch}"
+            << std::endl;
+
+  std::string agg;
+  std::cin >> agg;
```

Comment:
Could this be made a cmdline arg?

Reply:
Yes it can - I've made the changes and modified the README to reflect this change as well.

Reply:
Thanks. I think this way is simpler for human users as well as automation.